File: protocol/src/main/java/spp/protocol/processor/status/ProcessorConnection.java
package spp.protocol.processor.status;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.io.Serializable;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ProcessorConnection implements Serializable {

    private String processorId;
    private long connectionTime;

    public ProcessorConnection() {
    }

    public ProcessorConnection(String processorId, long connectionTime) {
        this.processorId = processorId;
        this.connectionTime = connectionTime;
    }

    public String getProcessorId() {
        return processorId;
    }

    @SuppressWarnings("unused")
    public void setProcessorId(String processorId) {
        this.processorId = processorId;
    }

    public long getConnectionTime() {
        return connectionTime;
    }

    @SuppressWarnings("unused")
    public void setConnectionTime(long connectionTime) {
        this.connectionTime = connectionTime;
    }
}

File: protocol/src/main/kotlin/spp/protocol/processor/status/ProcessorConnection.kt
package spp.protocol.processor.status

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.beans.ConstructorProperties
import java.io.Serializable

@JsonIgnoreProperties(ignoreUnknown = true)
data class ProcessorConnection @ConstructorProperties("processorId", "connectionTime", "meta") constructor(
    var processorId: String,
    var connectionTime: Long,
    var meta: MutableMap<String, Any> = mutableMapOf()
) : Serializable
